pkg/appgw/public_and_private_ip_test.go

```diff
@@ -333,208 +308,214 @@ var _ = Describe("Tests `appgw.ConfigBuilder`", func() {
-    --                    },
-    --                    "backendHttpSettings": {
-    --                        "id": "/subscriptions/--subscription--/resourceGroups/--resource-group--/providers/Microsoft.Network/applicationGateways/--app-gw-name--/backendHttpSettingsCollection/bp---namespace-----service-name---443-443-external-ingress-resource"
-    --                    },
-    --                    "httpListener": {
-    --                        "id": "/subscriptions/--subscription--/resourceGroups/--resource-group--/providers/Microsoft.Network/applicationGateways/--app-gw-name--/httpListeners/fl-443"
-    --                    },
-    --                    "ruleType": "Basic"
-    --                }
-    --            },
-    --            {
-    --                "etag": "*",
-    --                "id": "/subscriptions/--subscription--/resourceGroups/--resource-group--/providers/Microsoft.Network/applicationGateways/--app-gw-name--/requestRoutingRules/rr-80",
-    --                "name": "rr-80",
-    --                "properties": {
-    --                    "backendAddressPool": {
-    --                        "id": "/subscriptions/--subscription--/resourceGroups/--resource-group--/providers/Microsoft.Network/applicationGateways/--app-gw-name--/backendAddressPools/defaultaddresspool"
-    --                    },
-    --                    "backendHttpSettings": {
-    --                        "id": "/subscriptions/--subscription--/resourceGroups/--resource-group--/providers/Microsoft.Network/applicationGateways/--app-gw-name--/backendHttpSettingsCollection/bp---namespace-----service-name---80-80-internal-ingress-resource"
-    --                    },
-    --                    "httpListener": {
-    --                        "id": "/subscriptions/--subscription--/resourceGroups/--resource-group--/providers/Microsoft.Network/applicationGateways/--app-gw-name--/httpListeners/fl-80-privateip"
-    --                    },
-    --                    "ruleType": "Basic"
-    --                }
-    --            }
-    --        ],
-    --        "sslCertificates": [
-    --            {
-    --                "etag": "*",
-    --                "id": "/subscriptions/--subscription--/resourceGroups/--resource-group--/providers/Microsoft.Network/applicationGateways/--app-gw-name--/sslCertificates/--namespace-----the-name-of-the-secret--",
-    --                "name": "--namespace-----the-name-of-the-secret--",
-    --                "properties": {
-    --                    "data": "hhh",
-    --                    "password": "msazure"
-    --                }
-    --            }
-    --        ],
-    --        "urlPathMaps": null
-    --    },
-    --    "tags": {
-    --        "ingress-for-aks-cluster-id": "/subscriptions/subid/resourcegroups/aksresgp/providers/Microsoft.ContainerService/managedClusters/aksname",
-    --        "managed-by-k8s-ingress": "a/b/c"
-    --    }
-    --}`
+--    "properties": {
```

Comment:
The indentation changes make it really tough to see what changed in the output JSON.

Would you be able to either keep indentation as it was before this PR or first change the indentation in another PR and merge in here.  It'd be good to see what chunks of the App Gwy config actually changed.